common/unsafe/src/main/java/org/apache/spark/unsafe/memory/HeapMemoryAllocator.java

```diff
@@ -98,12 +100,13 @@ public void free(MemoryBlock memory) {
     long[] array = (long[]) memory.obj;
     memory.setObjAndOffset(null, 0);
 
-    if (shouldPool(size)) {
+    long alignedSize = ByteArrayMethods.roundNumberOfBytesToNearestWord(size);
```

Comment:
do we need to do it again here? I think the `size` is guaranteed to be aligned with word.